pennylane/templates/embeddings/basis.py

```diff
@@ -12,72 +12,62 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 r"""
-Contains the ``BasisEmbedding`` template.
+Contains the BasisEmbedding template.
 """
 # pylint: disable-msg=too-many-branches,too-many-arguments,protected-access
 import pennylane as qml
-from pennylane.templates.decorator import template
+from pennylane.operation import Operation, AnyWires
 from pennylane.wires import Wires
 
 
-def _preprocess(features, wires):
-    """Validate and pre-process inputs as follows:
-
-    * Check that the features tensor is one-dimensional.
-    * Check that the first dimension of the features tensor
-      has length :math:`n`, where :math:`n` is the number of qubits.
-    * Check that the entries of the features tensor are zeros and ones.
+class BasisEmbedding(Operation):
```

Comment:
I'm agnostic here, but if we decide to remove one or the other, lets turn it into an alias with a deprecation warning for one release cycle

Reply:
I would keep this duplication. At the moment they are doing the same, but an embedding and a state preparation could in the long term diverge. For example if users want us to add pre-processing because certain tricks emerge in research. 

Also it makes this embedding part of the embeddings library, which makes it easier to find.